x-pack/plugins/ml/public/application/jobs/components/custom_url_editor/utils.js

```diff
@@ -47,16 +48,9 @@ export function getNewCustomUrlDefaults(job, dashboards, indexPatterns) {
     datafeedConfig.indices !== undefined &&
     datafeedConfig.indices.length > 0
   ) {
-    const datafeedIndex = datafeedConfig.indices[0];
-    let defaultIndexPattern = indexPatterns.find((indexPattern) => {
-      return indexPattern.title === datafeedIndex;
-    });
-
-    if (defaultIndexPattern === undefined) {
-      defaultIndexPattern = indexPatterns[0];
-    }
-
-    kibanaSettings.discoverIndexPatternId = defaultIndexPattern.id;
+    const defaultIndexPatternId =
+      getIndexPatternIdFromName(datafeedConfig.indices.join()) || indexPatterns[0].id;
```

Comment:
we should use `??` for checks like this

Reply:
Fixed in https://github.com/elastic/kibana/pull/100789/commits/920f98c0c9fe6c0112e1edb32e875b6b4844e8b6